src/sagas/authorizationSaga.js

```diff
@@ -0,0 +1,101 @@
+import { Google } from 'expo';
+import { call, put, takeLatest } from 'redux-saga/effects';
+import { AsyncStorage } from 'react-native';
+import firebase from 'firebase';
+
+import actions from '../actions/index';
+import {
+  AUTHORIZE,
+  CHECK_IF_LOGGED_IN,
+} from '../constants/Actions';
+
+
+const isLoggedIn = () => firebase.auth().currentUser;
+
+const isUserEqual = (googleUser, firebaseUser) => {
+  if (firebaseUser) {
+    const { providerData } = firebaseUser;
+    for (let i = 0; i < providerData.length; i += 1) {
+      if (providerData[i].providerId === firebase.auth.GoogleAuthProvider.PROVIDER_ID
+        && providerData[i].uid === googleUser.getBasicProfile().getId()) {
+        return true;
+      }
+    }
+  }
+  return false;
+};
+
+const onSignIn = googleUser => {
+  // We need to register an Observer on Firebase Auth to make sure auth is initialized.
+  const unsubscribe = firebase.auth().onAuthStateChanged(firebaseUser => {
+    unsubscribe();
+    // Check if we are already signed-in Firebase with the correct user.
+    if (!isUserEqual(googleUser, firebaseUser)) {
```

Comment:
Flip the check condition and do a return inside instead of having this huge “if”